From 4bb2dccd69eaf8e25dd200f9a6deb883631db5c1 Mon Sep 17 00:00:00 2001
From: Igor Opaniuk <igor.opaniuk@toradex.com>
Date: Mon, 24 Jun 2019 11:39:14 +0300
Subject: [PATCH v2] apalis_imx6: increase phy autoneg timeout

Default value (4000ms) of PHY_ANEG_TIMEOUT for Micrel KSZ9031 contoller
isn't sufficient to finish auto-negotiation, which sometimes leads to
timeout errors:

Apalis iMX6 # dhcp
FEC Waiting for PHY auto negotiation to complete......... TIMEOUT !

Increase the auto-negotiation time-out to 15000ms.

Signed-off-by: Igor Opaniuk <igor.opaniuk@toradex.com>
---

v2:
- Based on empirical results from Philippe Schenker, increased timeout
  value to 15s, just to be on the safe side

 include/configs/apalis_imx6.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/apalis_imx6.h b/include/configs/apalis_imx6.h
index 4b0a3fb26b..df31cf6fe7 100644
--- a/include/configs/apalis_imx6.h
+++ b/include/configs/apalis_imx6.h
@@ -58,6 +58,7 @@
 #define IMX_FEC_BASE			ENET_BASE_ADDR
 #define CONFIG_FEC_XCV_TYPE		RGMII
 #define CONFIG_ETHPRIME			"FEC"
+#define PHY_ANEG_TIMEOUT		15000 /* PHY needs longer aneg time */
 #define CONFIG_FEC_MXC_PHYADDR		6
 #define CONFIG_TFTP_TSIZE
 
-- 
2.17.1
